src/async-syntax-highlighter.js

```diff
@@ -15,41 +17,84 @@ export default (options) => {
       return ReactAsyncHighlighter.loadAstGenerator();
     }
 
-    static registerLanguage = (_, language) => {
-      if (!supportsLanguageRegistering) {
+    static async loadLanguage(language) {
+      const languageLoader = languageLoaders[language];
+      if (typeof languageLoader === 'function') {
+        return languageLoader(ReactAsyncHighlighter.registerLanguage);
+      } else {
+        throw `Language ${language} not supported`
+      }
+    }
+    
+    static isRegistered = (language) => {
+      if(!registerLanguage) {
+        return true;
+      }
+
+      if (!ReactAsyncHighlighter.astGenerator) {
+        // Ast generator not available yet, but language will be registered once it is.
+        return ReactAsyncHighlighter.languages.findIndex(item => item.name === language) > -1;
+      }
+
+      return isLanguageRegistered(ReactAsyncHighlighter.astGenerator, language);
+    }
+
+    static registerLanguage = (name, language) => {
+      if (!registerLanguage) {
```

Comment:
it's my inclination to throw here as well

Reply:
Done